### PR TITLE
feat(ams-ui): crop cycle info page and other fixes

### PR DESCRIPTION
**Pull request by aashnaPrakash**
- cropCycleInfo page 

-  introduces a new menu component to show context menu on table rows in case the number of actions are large

- in case there are only two actions - like edit and delete, they are directly shown in the table

-  adapts the tables in fieldInfo, cropInfo, cropCycleInfo for the same

image

### CODE REVIEW

File: src/Pages/CropCycleInfoPage.js [hunk 1]
```diff
@@ -0,0 +1,327 @@
+import {
+  Button,
+  FormControl,
+  FormGroup,
+  InputLabel,
+  MenuItem,
+  Select,
+  Snackbar,
+  FormHelperText,
```
Comment (gautamguptabasant) [inline src/Pages/CropCycleInfoPage.js]: Organize imports. VS code will do it for you.

File: src/Pages/CropCycleInfoPage.js [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+
+/**
+ * Component to show Crop Cycle Information
+ * @returns form allowing location selection
+ * @returns form allowing field selection
+ * @returns Table to show crop cycle info
+ */
+function CropCycleInfoPage() {
+  const classes = useStyles();
+  const [locations, setLocations] = useState([]);
```
Comment (gautamguptabasant) [inline src/Pages/CropCycleInfoPage.js:102]: Please name this `locationCode`; that's what this stores.

**Review by gautamguptabasant — APPROVED** [review]
Looks good.
Can you attach some screen shots.

File: src/Pages/FieldInfoPage.js [hunk 3]
```diff
@@ -7,6 +7,7 @@ import {
   Snackbar,
   FormHelperText,
   Typography,
+  Button,
```
Comment (mahimavuppuluri) [inline src/Pages/FieldInfoPage.js]: please remove the fetch button, it's not needed anymore as location filtering is triggered by change in selection in the drop down. (Its removed already from master but this PR adds it again)

Comment (aashnaPrakash) [reply src/Pages/FieldInfoPage.js:10]: So the first time, when user lands on the page we are not showing any table?

File: src/Pages/CropCycleInfoPage.js (hunk 2 above)
Comment (aashnaPrakash) [reply src/Pages/CropCycleInfoPage.js:102]: It stores location objects - comprising both code and displayStr to be shown in dropdown.

File: src/Pages/CropCycleInfoPage.js [hunk 4]
```diff
@@ -0,0 +1,328 @@
+import {
+  Button,
+  FormControl,
+  FormGroup,
+
+  FormHelperText, InputLabel,
```
Comment (gautamguptabasant) [inline src/Pages/CropCycleInfoPage.js]: Formatting pls

File: src/Pages/CropCycleInfoPage.js (hunk 2 above)
Comment (gautamguptabasant) [reply src/Pages/CropCycleInfoPage.js:102]: Sorry, meant for locality

**Review by gautamguptabasant — APPROVED** [review]

**Comment by gautamguptabasant** [comment]
The menu icon (ellipsis) doesn't look wells spaced and is also not aligned in the center of the column. Please check.

File: src/Pages/CropCycleInfoPage.js (hunk 2 above)
Comment (aashnaPrakash) [reply src/Pages/CropCycleInfoPage.js:102]: done

File: src/Pages/CropCycleInfoPage.js (hunk 4 above)
Comment (aashnaPrakash) [reply src/Pages/CropCycleInfoPage.js]: done

**Comment by aashnaPrakash** [comment]
> The menu icon (ellipsis) doesn't look wells spaced and is also not aligned in the center of the column. Please check.

will take up menu action css in a follow up PR.

**Comment by aashnaPrakash** [comment]
> The menu icon (ellipsis) doesn't look wells spaced and is also not aligned in the center of the column. Please check.

done

**Pull request merged**